src/community/json-ld/src/main/java/org/geoserver/jsonld/expressions/JsonLdCqlManager.java

```diff
@@ -1,38 +1,125 @@
-/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
  * This code is licensed under the GPL 2.0 license, available at the root
  * application directory.
  */
 package org.geoserver.jsonld.expressions;
 
 import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import org.geoserver.jsonld.builders.impl.JsonBuilderContext;
-import org.geotools.data.complex.feature.type.ComplexFeatureTypeImpl;
 import org.geotools.factory.CommonFactoryFinder;
-import org.geotools.feature.type.Types;
+import org.geotools.filter.AttributeExpressionImpl;
 import org.geotools.filter.text.cql2.CQLException;
 import org.geotools.filter.text.ecql.ECQL;
-import org.opengis.feature.type.FeatureType;
+import org.geotools.filter.visitor.DuplicatingFilterVisitor;
+import org.opengis.filter.Filter;
 import org.opengis.filter.FilterFactory;
 import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Literal;
 import org.xml.sax.helpers.NamespaceSupport;
 
 /**
  * Helper class that mainly allows the extraction of CQL and Xpath expressions out of a plain text
- * string using special separators. It also provides some Utility methods to handle namespaces and
- * xpath syntax
+ * string using special separators. Moreover, since JsonLd templates can declare an xpath('some
+ * xpath') function that has no real FunctionExpression implementation, this class provides methods
+ * to extracts the xpath from the function as a literal, to substitute it after cql encoding
+ * happened with an AttributeExpression.
  */
-public class ExpressionsUtils {
+public class JsonLdCqlManager {
```

Comment:
```suggestion
public class JsonLdCQLManager {
```

Does the class need to be public?

Reply:
at the moment it is used outside its package by the builders